### PR TITLE
`bert_base_zh`, `bert_base_multi_cased`: Add BERT Base Variants

### PR DESCRIPTION
**Pull request by abheesht17**
Resolves https://github.com/keras-team/keras-nlp/issues/308

Converted model weights and vocab have been uploaded here: https://drive.google.com/drive/folders/19APUi7fobORdQjoe8YDyk8ou0x_L6hWu?usp=sharing.

Edit: Merging https://github.com/keras-team/keras-nlp/pull/320 with this PR. Might cause conflicts while merging with `master`, otherwise.

### CODE REVIEW

**Comment by mattdangerw** [comment]
@abheesht17 I think we would want to avoid the add_pooling_layer parameter if possible.

One thing we could check is the bert checkpoints on the original bert repo. Could you see if those include the pooling layer?

https://github.com/google-research/bert

**Comment by abheesht17** [comment]
> @abheesht17 I think we would want to avoid the add_pooling_layer parameter if possible.
> 
> One thing we could check is the bert checkpoints on the original bert repo. Could you see if those include the pooling layer?
> 
> https://github.com/google-research/bert

My bad. The pooling layers are present in this ckpt after all: https://storage.googleapis.com/tf_model_garden/nlp/bert/v3/multi_cased_L-12_H-768_A-12.tar.gz. Missed them earlier

File: tools/checkpoint_conversion/bert_base_multi_cased.ipynb [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+        "    )\n",
+        "    model.get_layer(f\"transformer_layer_{i}\")._feedforward_layernorm.gamma.assign(\n",
+        "        weights[f\"encoder/layer_with_weights-{i + 4}/_output_layer_norm/gamma/.ATTRIBUTES/VARIABLE_VALUE\"]\n",
+        "    )\n",
+        "    model.get_layer(f\"transformer_layer_{i}\")._feedforward_layernorm.beta.assign(\n",
+        "        weights[f\"encoder/layer_with_weights-{i + 4}/_output_layer_norm/beta/.ATTRIBUTES/VARIABLE_VALUE\"]\n",
+        "    )\n",
+        "\n",
+        "model.get_layer(\"pooled_dense\").kernel.assign(\n",
+        "    weights[f\"encoder/layer_with_weights-{i + 5}/kernel/.ATTRIBUTES/VARIABLE_VALUE\"]\n",
```
Comment (mattdangerw) [inline tools/checkpoint_conversion/bert_base_multi_cased.ipynb:589]: This is using a variable (i) that is not out of scope. Can't you just do num_layers + X?

Same comment for other colab.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
This looks good! We recently renamed the inputs to the models, could you make the changes in your colab to match?
https://github.com/keras-team/keras-nlp/pull/327

I think we are fine with the weights we have now (as the variable names do not depend on the input names), but might be good to double check the rename did not break the weights we uploaded.

**Comment by abheesht17** [comment]
> This looks good! We recently renamed the inputs to the models, could you make the changes in your colab to match? #327
> 
> I think we are fine with the weights we have now (as the variable names do not depend on the input names), but might be good to double check the rename did not break the weights we uploaded.

I think I already changed the names yesterday.

**Pull request merged**